File: Linked-List-I/038-linked-list-values/linked-list-values.js
// class Node {
//   constructor(val) {
//     this.val = val;
//     this.next = null;
//   }
// }

// O(n) time and O(n) space where n is the number of nodes
const linkedListValues = (head) => {
  let current = head;
  const values = [];

  while (current !== null) {
    values.push(current.val);
    current = current.next;
  }

  return values;
};

module.exports = {
  linkedListValues,
};

// Write a function, linkedListValues, that takes in the head of a linked list as an argument. The function should return an array containing all values of the nodes in the linked list.


// const a = new Node("a");
// const b = new Node("b");
// const c = new Node("c");
// const d = new Node("d");

// a.next = b;
// b.next = c;
// c.next = d;

// a -> b -> c -> d

// linkedListValues(a); // -> [ 'a', 'b', 'c', 'd' ]

//

File: Linked-List-I/041-linked-list-values/linked-list-values.py
# class Node:
#   def __init__(self, val):
#     self.val = val
#     self.next = None

# O(n) time and O(n) space
def linked_list_values(head):
  list_vals = []
  current = head

  while current is not None:
    list_vals.append(curent.val)
    current = current.next

  return list_vals
